File: Problems/RepeatedDNASequences.js
/**
 * @param {string} s
 * @return {string[]}
 */
var findRepeatedDnaSequences = function(s) {
    let current = s.slice(0,10);
    const seen = new Set([current]);
    const result = new Set();
    
    for(let i = 10; i < s.length; i++){
        current = current.slice(1) + s[i];
        if(seen.has(current)) result.add(current);
        seen.add(current);
    }
    return [...result];
};

File: ProgrammingSkills/RepeatedDNASequences.py
class Solution:
    def findRepeatedDnaSequences(self, s: str) -> List[str]:
        result = []
        aux = set()
        
        i, j = 0, 9
        
        while j < len(s):
          
            if s[i:j+1] in aux:
                if s[i:j+1] not in result:
                    result.append(s[i:j+1])                    
                i += 1
                j += 1
                
            elif s[i:j+1] not in aux:
                aux.add(s[i:j+1])
                i += 1
                j += 1
        
        return result
